src/components/product-card/ProductCard.js

```diff
@@ -35,13 +43,20 @@ export default function ProductCard({ product }) {
           )}
 
           <div className="card__name prod__card-name">
-            <p className="card__charc-name prod__charc-name">{product.name}</p>
+            <p className="card__charc-name prod__charc-name" onClick = {testFunc}>{product.name}</p>
             <p className="card__charc-series prod__charc-series">
               ₹ {product.price}
             </p>
-            <button className="btn solid__secondry prod__card-btn">
+            {cartProducts.find((cartProd) => cartProd._id === product._id) ? (<>
+              <button className="btn solid__primary prod__card-btn" onClick={() => cartDispatch({type : "REMOVE_FROM_CART", payload : product})}>
+              Remove From Cart
+            </button>
+            </>) : (<>
```

Comment:
using fragments (<>) tag can be avoided if there is only one element to be rendered. The fragment is generally used when there are multiple elements and we need one parent to wrap it.